src/dlls/mscoree/mscoree.cpp

```diff
@@ -907,9 +907,23 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength
     if (pBuffer == NULL)
         IfFailGo(E_POINTER);
 
+#ifdef CROSSGEN_COMPILE
```

Comment:
The original code calls PAL_GetPALDirectory, which returns the path to libcoreclr.so (not the path to PAL, as the name might suggest). Since crossgen doesn’t load libcoreclr.so, PAL_GetPALDirectory fails.

From: Jan Kotas [mailto:notifications@github.com]
Sent: Tuesday, April 28, 2015 3:14 PM
To: dotnet/coreclr
Cc: John Chen (CLR)
Subject: Re: [coreclr] Build crossgen for Linux (#860)


In src/dlls/mscoree/mscoree.cpp<https://github.com/dotnet/coreclr/pull/860#discussion_r29294169>:

> @@ -907,9 +907,23 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength

>      if (pBuffer == NULL)

>          IfFailGo(E_POINTER);

>

> +#ifdef CROSSGEN_COMPILE

I do not see why we need this ifdef - could you please shed some light on it?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/860/files#r29294169>.